client-js/main/client/client-request.js

```diff
@@ -0,0 +1,600 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+"use strict";
+
+import {Message} from 'google-protobuf';
+import {CompositeFilter, Filter} from '../proto/spine/client/filters_pb';
+import {OrderBy} from '../proto/spine/client/query_pb';
+import {Command, CommandId} from '../proto/spine/core/command_pb';
+import {MessageId, Origin} from '../proto/spine/core/diagnostics_pb';
+import {AnyPacker} from "./any-packer";
+import {Filters} from "./actor-request-factory";
+import {Type} from "./typed-message";
+
+/**
+ * @typedef EventSubscriptionCallbacks
+ *
+ * A pair of callbacks that allow to add an event consumer and cancel the subscription
+ * respectively.
+ *
+ * @property {consumerCallback<eventConsumer>} subscribe the callback which allows to setup an
+ *                                                       event consumer to use for the subscription
+ * @property {parameterlessCallback} unsubscribe the callback which allows to cancel the
+ *                                               subscription
+ */
+
+/**
+ * A request from a client to the Spine backend.
+ *
+ * @abstract
+ */
+class ClientRequest {
```

Comment:
Could you split this file up? It's LoC count is at 600 already. GitHub even considers it too large to display.